### PR TITLE
Build setup.exe in temp and copy to destination

### PR DESCRIPTION
**Pull request by easel**
Prevents users from reading a partially written setup.exe.

### CODE REVIEW

File: src/Update/Program.cs [hunk 1]
```diff
@@ -458,16 +458,26 @@ public void Releasify(string package, string targetDir = null, string packagesDi
...
+            var targetSetupExe = Path.Combine(di.FullName, "Setup.exe");
+            if(File.Exists(targetSetupExe))
+            {
+                File.Delete(targetSetupExe);
             }
+            File.Move(tempTargetSetupExe, targetSetupExe);
 
-            if (generateMsi) {
+            if (generateMsi)
+            {
```
Comment (paulcbetts) [inline src/Update/Program.cs]: y tho

**Comment by easel** [comment]
Meh. Sorry about that. Totally blaming visual studio for that one. Tightened up the other trailing braces as well.

**Comment by paulcbetts** [comment]
@easel Can you tell me more about this fix? What is it trying to solve?

**Comment by easel** [comment]
It's trying to solve the problem of building setup files directly to a published location. With the current code, if clients try to run the installer while the build is running, they get an invalid file. With this fix, the file is generated offline and replaced atomically so clients never see an invalid file.

**Comment by robmen** [comment]
The problem with using %TMP% is that virus checkers often get huffy about writing temporary .exes then moving them around. It'd be better if we standardize on an `obj` folder that is relative to the build process (just like VS does for projects).

**Pull request closed without merging**